microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -85,119 +100,267 @@ type RouterCfg struct {
 	GRAPH_JSON string
 }
 
-func getManifestYaml(step string) string {
-	var tmpltFile string
-	//TODO add validation to rule out unexpected case like both embedding and retrieving
-	if step == Embedding {
-		tmpltFile = yaml_dir + embedding_yaml
-	} else if step == TeiEmbedding {
-		tmpltFile = yaml_dir + tei_embedding_service_yaml
-	} else if step == TeiEmbeddingGaudi {
-		tmpltFile = yaml_dir + tei_embedding_gaudi_service_yaml
-	} else if step == VectorDB {
-		tmpltFile = yaml_dir + redis_vector_db_yaml
-	} else if step == Retriever {
-		tmpltFile = yaml_dir + retriever_yaml
-	} else if step == Reranking {
-		tmpltFile = yaml_dir + reranking_yaml
-	} else if step == TeiReranking {
-		tmpltFile = yaml_dir + tei_reranking_service_yaml
-	} else if step == Tgi {
-		tmpltFile = yaml_dir + tgi_service_yaml
-	} else if step == TgiGaudi {
-		tmpltFile = yaml_dir + tgi_gaudi_service_yaml
-	} else if step == Llm {
-		tmpltFile = yaml_dir + llm_yaml
-	} else if step == DocSum {
-		tmpltFile = yaml_dir + docsum_llm_yaml
-	} else if step == DocSumGaudi {
-		tmpltFile = yaml_dir + docsum_gaudi_llm_yaml
-	} else if step == Router {
-		tmpltFile = yaml_dir + gmc_router_yaml
+func lookupManifestDir(step string) string {
+	value, exist := yamlDict[step]
+	if exist {
+		return value
 	} else {
 		return ""
 	}
-	return tmpltFile
 }
 
-func reconcileResource(ctx context.Context, client client.Client, step string, ns string, svc string, svcCfg *map[string]string, retSvc *corev1.Service) error {
-	fmt.Printf("get step %s config for %s@%s: %v\n", step, svc, ns, svcCfg)
-	tmpltFile := getManifestYaml(step)
-	if tmpltFile == "" {
-		return errors.New("unexpected target")
+// func getManifestYaml(step string) string {
```

Comment:
@KfreeZ  can these code lines be removed if they are not used?

Reply:
yes, I'm on it